src/Http/Headers/test/MediaTypeHeaderValueTest.cs

```diff
@@ -688,6 +688,100 @@ public void TryParseStrictList_WithSomeInvalidValues_ReturnsFalse()
             Assert.False(MediaTypeHeaderValue.TryParseStrictList(inputs, out results));
         }
 
+        [Theory]
+        [InlineData("*/*;", "*/*")]
+        [InlineData("text/*", "text/*")]
+        [InlineData("text/*;", "*/*")]
+        [InlineData("text/plain;", "text/plain")]
+        [InlineData("text/plain", "text/*")]
+        [InlineData("text/plain;", "*/*")]
+        [InlineData("*/*;missingparam=4", "*/*")]
+        [InlineData("text/*;missingparam=4;", "*/*;")]
+        [InlineData("text/plain;missingparam=4", "*/*;")]
+        [InlineData("text/plain;missingparam=4", "text/*")]
+        [InlineData("text/plain;charset=utf-8", "text/plain;charset=utf-8")]
+        [InlineData("text/plain;version=v1", "Text/plain;Version=v1")]
+        [InlineData("text/plain;version=v1", "tExT/plain;version=V1")]
+        [InlineData("text/plain;version=v1", "TEXT/PLAIN;VERSION=V1")]
+        [InlineData("text/plain;charset=utf-8;foo=bar;q=0.0", "text/plain;charset=utf-8;foo=bar;q=0.0")]
+        [InlineData("text/plain;charset=utf-8;foo=bar;q=0.0", "text/plain;foo=bar;q=0.0;charset=utf-8")] // different order of parameters
+        [InlineData("text/plain;charset=utf-8;foo=bar;q=0.0", "text/*;charset=utf-8;foo=bar;q=0.0")]
+        [InlineData("text/plain;charset=utf-8;foo=bar;q=0.0", "*/*;charset=utf-8;foo=bar;q=0.0")]
+        [InlineData("application/json;v=2", "application/json;*")]
+        [InlineData("application/json;v=2;charset=utf-8", "application/json;v=2;*")]
+        public void MatchesMediaType_PositiveCases(string mediaType1, string mediaType2)
```

Comment:
How much does ordering matter here? 

Take this scenario we're trying to replace:
https://github.com/grpc/grpc-dotnet/blob/3ce9b104524a4929f5014c13cd99ba9a1c2431d4/src/Shared/CommonGrpcProtocolHelpers.cs#L26
https://github.com/grpc/grpc-dotnet/search?q=IsContentType&unscoped_q=IsContentType
Or here:
https://github.com/microsoft/reverse-proxy/blob/419c4d6502d1e8510a6dd0a0bedd67fa410ec1f2/src/ReverseProxy/Utilities/ProtocolHelper.cs#L30

You'd want one pre-parsed MediaTypeHeaderValue that you're comparing all incoming headers against. If there's a wildcard, it's always in the original value, not in the second value you're comparing to.

Reply:
So, MediaTypeHeaderValue.Parse("*/*").MatchesMediaType("text/plain") should return true and MediaTypeHeaderValue.Parse("text/plain").MatchesMediaType("*/*") should return false right?

Reply:
Yes, I'm still interested in contributing. It looks like I need to make some changes to invert comparisons and remove unnecessary test cases. I'll let you know when I finish the implementation.

Reply:
Can you take a look at it again? @Tratcher